api/nodes/serializers.py

```diff
@@ -75,6 +76,20 @@ def update_institutions(node, new_institutions, user, post=False):
         node.add_affiliated_institution(inst, user)
 
 
+class RegionRelationshipField(RelationshipField):
+
+    def get_object(self, region_id):
+        try:
+            region = Region.objects.get(_id=region_id)
+        except Region.DoesNotExist:
+            raise exceptions.ValidationError(detail='Region {} is invalid.'.format(region_id))
+        return region
+
+    def to_internal_value(self, data):
+        region = self.get_object(data)
```

Comment:
Minor: Is `get_object` necessary?

```python

def to_internal_value(...):
    try:
        region_id = Region.objects.filter(_id=data).values_list('id', flat=True).get()
    except Region.DoesNotExist:
        raise ValidationError(...)
    return {'region_id': region_id}
```

Reply:
Hmm, doesn't seem like it.

We do this for some writeable relationships:
https://github.com/CenterForOpenScience/osf.io/blob/75307d72cdfe5b1870674dac38e242883252b16b/api/preprints/serializers.py#L26-L48

but not for others:
https://github.com/CenterForOpenScience/osf.io/blob/75307d72cdfe5b1870674dac38e242883252b16b/api/preprints/serializers.py#L51-L56

Do you have a preference?

Reply:
Ah, I see--I hadn't seen PrimaryFileRelationshipField et al.; thanks. I think if you're only going to serialize to `{'region_id': ...}`, then `get_object` is overkill (since you don't actually need the whole object). If you're going to serialize to `{'region': region_obj}`, then `get_object` makes more sense.

Reply:
Ah, good point. I'll switch that around.